packages/react/src/components/Authenticator/Authenticator.tsx

```diff
@@ -96,6 +101,13 @@ export function AuthenticatorInternal({
  * [📖 Docs](https://ui.docs.amplify.aws/react/connected-components/authenticator)
  */
 export function Authenticator(props: AuthenticatorProps): JSX.Element {
+  React.useEffect(() => {
```

Comment:
Technically, this will run multiple times if this component remounts. Could that possibly mess up the data?

Reply:
My understanding is this just sets up the correct header to be included with Cognito API calls, so it shouldn't have an negative impact.

Reply:
Oh yeah, I was worried this would add duplicate headers. But looks like Amplify JS handles duplicate for us! ([src](https://github.com/aws-amplify/amplify-js/blob/190bd501236a59b16318802b51a0494f7374d299/packages/amazon-cognito-identity-js/src/UserAgent.js#L13))

Reply:
Just tested this by running our Next example app and adding a button to conditionally render the Authenticator, toggled it a few times, and tried to log in. The UA string seemed fine.